### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by Mikey1-nuel**
## Blog app: add API endpoints

**In this project I:**

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should **not** be set by passing a parameter.
- Add an API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -77,6 +77,8 @@ gem 'devise'
 
 gem 'cancancan'
```
Comment (bobb-Rob) [inline Gemfile:79]: - [ ] Your effort in creating the migrations to add references to the post, comment, and likes tables is commendable. However, it seems there is an issue with the current order of your migration files. The migration process in Ruby starts from the first migration and proceeds in chronological order. The first migration you have is attempting to add a reference to the post table, but the post table has not been created yet.

  To resolve this issue, it's essential to ensure that all table creations are completed before any alterations that involve adding references to those tables. This way, the necessary tables will exist in the database before attempting to establish foreign key relationships.

  **To fix the problem, follow these steps:**
  In the image below, all the migrations that I label "A" Should come before all those I label with "B".

  **_Create Migrations for Table Creation:_** Ensure that you have migration files to create the post, comment, and likes tables before any migrations that involve adding references to them.

image

Comment (bobb-Rob) [inline Gemfile:79]: As a result of the wrong order of migration files, I was unable to test your API, the next reviewer will test it and give appropriate feedback. 

I wish you the best.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @Mikey1-nuel, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ All linters checks are ok
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @Mikey1-nuel , 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Readme file, API endpoints
✅ ... Green linters 

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**